### ISSUE TITLE
OSError when trying to start files server

### ISSUE THREAD

**Issue by Deng-Yao** [spyder-ide/spyder#18262]
### What steps will reproduce the problem?

？？？???????????????????????????????????????????????

### Traceback
```python-traceback
  File "D:\Anaconda3\lib\threading.py", line 932, in _bootstrap_inner
    self.run()
  File "D:\Anaconda3\lib\threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "D:\Anaconda3\lib\site-packages\spyder\app\mainwindow.py", line 3168, in start_open_files_server
    self.open_files_server.bind(('127.0.0.1', port))
OSError: [WinError 10013] 以一种访问权限不允许的方式做了一个访问套接字的尝试。
WARNING:traitlets:kernel restarted
WARNING:traitlets:kernel restarted
WARNING:traitlets:kernel restarted
WARNING:traitlets:kernel restarted
WARNING:traitlets:KernelRestarter: restart failed
WARNING:traitlets:kernel died: 25.0005099773407
WARNING:traitlets:kernel died: 25.00770354270935
```

## Versions

* Spyder version: 4.2.5 
* Python version: 3.8.8
* Qt version: 5.9.7
* PyQt5 version: 5.9.2
* Operating System: Windows 10

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0            :  1.4.0 (OK)
chardet >=2.0.0                 :  4.0.0 (OK)
cloudpickle >=0.5.0             :  1.6.0 (OK)
diff_match_patch >=20181111     :  20200713 (OK)
intervaltree >=3.0.2            :  3.1.0 (OK)
IPython >=7.6.0                 :  7.22.0 (OK)
jedi =0.17.2                    :  0.17.2 (OK)
jsonschema >=3.2.0              :  3.2.0 (OK)
keyring >=17.0.0                :  22.3.0 (OK)
nbconvert >=4.0                 :  6.0.7 (OK)
numpydoc >=0.6.0                :  1.1.0 (OK)
paramiko >=2.4.0                :  2.7.2 (OK)
parso =0.7.0                    :  0.7.0 (OK)
pexpect >=4.4.0                 :  4.8.0 (OK)
pickleshare >=0.4               :  0.7.5 (OK)
psutil >=5.3                    :  5.8.0 (OK)
pygments >=2.0                  :  2.8.1 (OK)
pylint >=1.0                    :  2.7.4 (OK)
pyls >=0.36.2;<1.0.0            :  0.36.2 (OK)
pyls_black >=0.4.6              :  0.4.6 (OK)
pyls_spyder >=0.3.2             :  0.3.2 (OK)
qdarkstyle >=2.8;<3.0           :  2.8.1 (OK)
qtawesome >=0.5.7               :  1.0.2 (OK)
qtconsole >=5.0.3               :  5.0.3 (OK)
qtpy >=1.5.0                    :  1.9.0 (OK)
rtree >=0.8.3                   :  0.9.7 (OK)
setuptools >=39.0.0             :  52.0.0.post20210125 (OK)
sphinx >=0.6.6                  :  4.0.1 (OK)
spyder_kernels >=1.10.2;<1.11.0 :  1.10.2 (OK)
textdistance >=4.2.0            :  4.2.1 (OK)
three_merge >=0.1.1             :  0.1.1 (OK)
watchdog >=0.10.3;<2.0.0        :  1.0.2 (OK)
zmq >=17                        :  20.0.0 (OK)

# Optional:
cython >=0.21                   :  0.29.23 (OK)
matplotlib >=2.0.0              :  3.3.4 (OK)
numpy >=1.7                     :  1.22.2 (OK)
pandas >=1.1.1                  :  1.1.3 (OK)
scipy >=0.17.0                  :  1.7.1 (OK)
sympy >=0.7.3                   :  1.8 (OK)
```

**Comment by athompson673** [comment]
you are evidently running quite an old version of spyder itself, as well as a number of libraries. Please try updating to a newer version. With the number of things out-of-date, I would personally recommend you install a totally new version of python from python.org and install spyder from scratch via `pip`. This way everything will be the most recent version available from pypi. If you have a strong reason to keep using the old installation of python, you _can_ run spyder on the new python, while executing code with the old version, but it's never a bad idea to try and make your code run on the newest version.

**Comment by Deng-Yao** [comment]
这是来自QQ邮箱的自动回复邮件。您好，我已收到您的邮件。

**Comment by ccordoba12** [comment]
Hey @Deng-Yao, thanks for reporting. It seems your antivirus or firewall doesn't allow to open ports and Spyder crashes due to that.

We'll fix that in our next version (5.3.2), to be released at the beginning of July. For now, please open the Anaconda Prompt and run there

    spyder --new-instance

Afterwards, please go to the menu

    Tools > Preferences > Application > Advanced settings

and disable the option called

    Use a single instance

After that you can close Spyder and start it again as you normally do and shouldn't see any issues.

**Comment by Deng-Yao** [comment]
Thank you for your help. The problem was solved after I restarted my computer.

**Comment by Deng-Yao** [comment]
> 嘿@Deng-Yao，感谢报道。您的防病毒软件或防火墙似乎不允许打开端口，因此 Spyder 会崩溃。
> 
> 我们将在 7 月初发布的下一个版本 (5.3.2) 中解决这个问题。现在，请打开 Anaconda Prompt 并在那里运行
> 
> ```
> spyder --new-instance
> ```
> 
> 之后，请前往菜单
> 
> ```
> Tools > Preferences > Application > Advanced settings
> ```
> 
> 并禁用名为
> 
> ```
> Use a single instance
> ```
> 
> 之后，您可以像往常一样关闭 Spyder 并重新启动它，应该不会看到任何问题。



> you are evidently running quite an old version of spyder itself, as well as a number of libraries. Please try updating to a newer version. With the number of things out-of-date, I would personally recommend you install a totally new version of python from python.org and install spyder from scratch via `pip`. This way everything will be the most recent version available from pypi. If you have a strong reason to keep using the old installation of python, you _can_ run spyder on the new python, while executing code with the old version, but it's never a bad idea to try and make your code run on the newest version.

Thank you for your help. The problem was solved after I restarted my computer.

**Comment by Deng-Yao** [comment]
> you are evidently running quite an old version of spyder itself, as well as a number of libraries. Please try updating to a newer version. With the number of things out-of-date, I would personally recommend you install a totally new version of python from python.org and install spyder from scratch via `pip`. This way everything will be the most recent version available from pypi. If you have a strong reason to keep using the old installation of python, you _can_ run spyder on the new python, while executing code with the old version, but it's never a bad idea to try and make your code run on the newest version.

Thank you for your help. The problem was solved after I restarted my computer.

**Comment by Deng-Yao** [comment]
> Hey @Deng-Yao, thanks for reporting. It seems your antivirus or firewall doesn't allow to open ports and Spyder crashes due to that.
> 
> We'll fix that in our next version (5.3.2), to be released at the beginning of July. For now, please open the Anaconda Prompt and run there
> 
> ```
> spyder --new-instance
> ```
> 
> Afterwards, please go to the menu
> 
> ```
> Tools > Preferences > Application > Advanced settings
> ```
> 
> and disable the option called
> 
> ```
> Use a single instance
> ```
> 
> After that you can close Spyder and start it again as you normally do and shouldn't see any issues.

Thank you for your help. The problem was solved after I restarted my computer.
